pydantic_ai_slim/pydantic_ai/_function_schema.py

```diff
@@ -27,24 +32,53 @@
 __all__ = ('function_schema',)
 
 
-class FunctionSchema(TypedDict):
+@dataclass
+class FunctionSchema:
     """Internal information about a function schema."""
 
+    function: Callable[..., Any]
```

Comment:
I feel like the intent of this was to be serializable etc., I would guess inspired by pydantic-ai. I don't necessarily think it's a problem to change, but I feel like we should confirm with Samuel that this change makes sense.